src/services/expense.service.js

```diff
@@ -0,0 +1,74 @@
+const { Op } = require('sequelize');
+const { Expense } = require('../models/Expense.model.js');
+
+const getAllExpense = async ({ userId, categories, from, to }) => {
+  const params = {};
+
+  if (userId) {
+    params.userId = userId;
+  }
+
+  if (categories) {
+    if (Array.isArray(categories)) {
+      params.category = { [Op.in]: categories };
+    } else {
+      params.category = categories;
+    }
+  }
+
+  if (from || to) {
+    params.spentAt = {};
+
+    if (from) {
+      params.spentAt[Op.gte] = from;
+    }
+
+    if (to) {
+      params.spentAt[Op.lte] = to;
+    }
+  }
+
+  const expList = Expense.findAll({ where: params });
+
+  return expList;
+};
+
+const getExpenseById = async (id) => {
+  return Expense.findByPk(id);
+};
+
+const createExpense = async (data) => {
+  return Expense.create(data);
+};
+
+const updateExpense = async (
+  id,
+  { spentAt, title, amount, category, note },
+) => {
+  await Expense.update(
+    {
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    },
+    { where: { id } },
+  );
```

Comment:
Ensure that the update operation handles cases where the provided fields might be undefined. You might want to filter out undefined fields before updating to avoid overwriting existing data with undefined values.